cluster-operator/src/test/java/io/strimzi/operator/cluster/model/EntityUserOperatorTest.java

```diff
@@ -163,4 +166,37 @@ public void testGetContainers() {
         assertEquals("/opt/user-operator/custom-config/", container.getVolumeMounts().get(0).getMountPath());
         assertEquals("entity-user-operator-metrics-and-logging", container.getVolumeMounts().get(0).getName());
     }
+
+    @Test
+    public void testFromCrdCaValidityAndRenewal() {
+        EntityUserOperatorSpec entityUserOperatorSpec = new EntityUserOperatorSpecBuilder()
+                .build();
+        EntityOperatorSpec entityOperatorSpec = new EntityOperatorSpecBuilder()
+                .withUserOperator(entityUserOperatorSpec)
+                .build();
+        CertificateAuthority ca = new CertificateAuthority();
+        ca.setValidityDays(42);
+        ca.setRenewalDays(69);
+        Kafka customValues =
+                new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image, healthDelay, healthTimeout))
+                        .editSpec()
+                        .withEntityOperator(entityOperatorSpec)
+                        .withClientsCa(ca)
+                        .endSpec()
+                        .build();
+        EntityUserOperator entityUserOperator = EntityUserOperator.fromCrd(customValues);
+
+        Kafka defaultValues =
+                new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas, image, healthDelay, healthTimeout))
+                        .editSpec()
+                        .withEntityOperator(entityOperatorSpec)
+                        .endSpec()
+                        .build();
+        EntityUserOperator entityUserOperator2 = EntityUserOperator.fromCrd(defaultValues);
+
+        assertEquals(42, entityUserOperator.getClientsCaValidityDays());
+        assertEquals(69, entityUserOperator.getClientsCaRenewalDays());
+        assertEquals(AbstractModel.CERTS_VALIDITY_DAYS, entityUserOperator2.getClientsCaValidityDays());
+        assertEquals(AbstractModel.CERTS_RENEWAL_DAYS, entityUserOperator2.getClientsCaRenewalDays());
+    }
```

Comment:
What I don't see it a test covering the end-to-end scenario: That the UO `Deployment` ends up with the relevent env vars set.